roles/ddns_updater/tasks/main.yml

```diff
@@ -0,0 +1,46 @@
+---
+- name: Create DDNS Updater Directories
+  file:
+    path: "{{ ddns_updater_data_directory }}"
+    state: directory
+
+- name: Check config is defined
+  fail: msg="No 'ddns_updater_config' defined."
+  when: ddns_updater_config is not defined
+
+- name: Generate config.json
+  copy:
+    content: "{{ ddns_updater_config | to_nice_json }}"
+    dest: "{{ ddns_updater_data_directory }}/config.json"
+
+- name: Generate dns docker environment
+  set_fact:
+    ddns_update_env: |
+      TZ: "{{ ansible_nas_timezone }}"
+      {% if ddns_updater_gotify_token is defined %}
+      GOTIFY_TOKEN: {{ ddns_updater_gotify_token }}
+      GOTIFY_URL: {{ gotify_url }}
+      {% endif %}
+      {% for env in ddns_update_extra_env|dict2items %}
+      {{ env.key }}: {{ env.value }}
+      {% endfor %}
```

Comment:
I put some thought into how to make it configurable. Perhaps thats also an option for other roles as well?
With this one could also exchange the docker image and use one with slightly different env variables without need to modify the role itself.

Reply:
I think the gotify section is enough here, but could it also be moved straight to the `docker_container` block within the task to save on the extra config complication? If you feel there are other config options that should be defined, include them too and simply create a variable, rather than trying to be overly clever with mapping keys and values to env vars.

Reply:
I get what you mean. I tried to simplify the task and put everything directly in the env of the container.
the only downside is that the container will now try to contact the gotify server with the invalid token from the default and fail, but thats only an error log in the container an is probably ok.